src/app/destiny2/d2-definitions.ts

```diff
@@ -155,11 +161,20 @@ async function getDefinitionsUncached() {
   lazyTables.forEach((tableShort) => {
     const table = `Destiny${tableShort}Definition`;
     defs[tableShort] = {
-      get(id: number) {
+      get(id: number, requestor?: any) {
         const dbTable = db[table];
         if (!dbTable) {
           throw new Error(`Table ${table} does not exist in the manifest`);
         }
+        if (!dbTable[id]) {
+          const requestingEntryInfo =
+            typeof requestor === 'object' ? requestor.hash : String(requestor);
+          reportException('hashLookupFailure', new Error('hash lookup failure'), {
```

Comment:
Mildly worried that this'll spam us from places where it's OK for the definition to be missing?

Reply:
ideally, those shouldn't really exist or our `get()` typings are wrong. the purpose of lookup hashes is to be internally consistent.

wishlists i guess being a key exception. maybe? do we do lookups of those items or just check hashes? if we do i will suppress it for wishlists, but those aside, i would love to know where we are regularly or even purposefully failing lookups, so we know what might need attention.

Reply:
removed some intentional lookup fails, we don't do item def lookups on user input, (wishlists), need to keep an eye on `SeasonalRank.tsx` (`season.artifactItemHash || 0`) but worst case that failed lookup will generate some sentry noise if they unexpectedly remove seasonal artifacts and we haven't programmed a fallback yet